neo/rawio/mearecrawio.py

```diff
@@ -36,52 +48,74 @@ class MEArecRawIO(BaseRawIO):
     extensions = ['h5']
     rawmode = 'one-file'
 
-    def __init__(self, filename=''):
+    def __init__(self, filename='', load_spiketrains=True, load_recordings=True):
         BaseRawIO.__init__(self)
         self.filename = filename
-
+        self.load_spiketrains = load_spiketrains
+        self.load_recordings = load_recordings
+        
     def _source_name(self):
         return self.filename
 
     def _parse_header(self):
+        load = ["channel_positions"] 
+        if self.load_recordings:
+            load.append("recordings")
+        if self.load_spiketrains:
+            load.append("spiketrains")
+        
         import MEArec as mr
         self._recgen = mr.load_recordings(recordings=self.filename, return_h5_objects=True,
                                           check_suffix=False,
-                                          load=['recordings', 'spiketrains', 'channel_positions'],
+                                          load=load,
                                           load_waveforms=False)
-        self._sampling_rate = self._recgen.info['recordings']['fs']
-        self._recordings = self._recgen.recordings
-        self._num_frames, self._num_channels = self._recordings.shape
-
-        signal_streams = np.array([('Signals', '0')], dtype=_signal_stream_dtype)
 
+        self.info_dict = deepcopy(self._recgen.info)
+        self.channel_positions = self._recgen.channel_positions
+        if self.load_recordings:
+            self._recordings = self._recgen.recordings
+        if self.load_spiketrains:
+            self._spiketrains = self._recgen.spiketrains
+
+        self._sampling_rate = self.info_dict['recordings']['fs']
+        self.duration_seconds = self.info_dict["recordings"]["duration"]
+        self._num_frames = int(self._sampling_rate * self.duration_seconds)
+        self._num_channels = self.channel_positions.shape[0]
+        self._dtype = self.info_dict["recordings"]["dtype"]
+        
+        signals = [('Signals', '0')] 
+        signal_streams = np.array(signals, dtype=_signal_stream_dtype)
```

Comment:
With load_recording=False this shoudl be empty.

Reply:
I tried this, but then if someone calls `get_analog_signal_chunk` in that configuration it will throw a criptic error.

```
    def _get_stream_index_from_arg(self, stream_index_arg):
        if stream_index_arg is None:
>           assert self.header['signal_streams'].size == 1
E           AssertionError
```

This is generated because the general rawio. calls:

https://github.com/h-mayorquin/python-neo/blob/ac508e117097f44ea82b4058c4a4d4f6d1e7c2eb/neo/rawio/baserawio.py#L531-L553

Which in turns call `get_stream_index_from_arg` which then asks for a header.

If we only get rid of the channels we still get an error:

```
        if signal_streams.size > 0:
>           assert signal_channels.size > 0, 'Signal stream but no signal_channels!!!'
E           AssertionError: Signal stream but no signal_channels!!!
```

I prefer to have informative error messages but I don't know the cost of keeping those references open. What do you think?